### DISCUSSION TITLE
Trouble talking to MySQL 8

### DISCUSSION THREAD

**Discussion by matkey** [nextgenhealthcare/connect#4873] [Q&A]
Evening all,

I am having trouble getting Mirth Connect to talk to MySQL 8. I am trying to write to a database. Code is fine on MySQL 5.7.28

Running Mirth 3.12 in Docker.
Running MySQL 8.0.26 in Docker

Database Driver is com.mysql.cj.jdbc.Driver
URL is jdbc:mysql://192.168.0.12:3306/patient_hl7?allowPublicKeyRetrieval=true&sslMode=D ISABLED
I am usin the Get Tables trick to make sure I can access the database. This works fine on MySQL 5.7

However if I run a MySQL 8.0.26 docker image using the same connection string then I dont get an error connecting to it, but nothing is returned in the Get Tables view.
I know the "allowPublicKeyRetrieval" bit is needed as without it, I get an exception in Mirth Connect. We also use that on our SQL connection string from within our own product to talk to the database in C#. Basically, I know the URL is correct.

Works on MySQL 5.7 but nothing returned on MySQL 8.

Any ideas where I can start looking?
Are there log files in Mirth somewhere in the Docker image? (I am new to Mirth - taking over an existing system from another developer)

Thanks in advance

Mat

**Comment by kpalang** [comment]
Was your MySql 5 also running on Docker? Also, if your containers are running on the same host, I would use Docker's DNS system instead of hardcoding the IP address in jdbc string.

For Mirth logs do `docker logs <your-mirth-service-name>`

**Comment by matkey** [comment]
Nothing in the logs looks out of place.....

INFO  2021-11-23 21:13:39,635 [Main Server Thread] com.mirth.connect.server.Mirth: Mirth Connect 3.12.0 (Built on September 2, 2021) server successfully started.
INFO  2021-11-23 21:13:39,639 [Main Server Thread] com.mirth.connect.server.Mirth: This product was developed by NextGen Healthcare (https://www.nextgen.com) and its contributors (c)2005-2021.
INFO  2021-11-23 21:13:39,639 [Main Server Thread] com.mirth.connect.server.Mirth: Running OpenJDK 64-Bit Server VM 11.0.12 on Linux (5.10.60.1-microsoft-standard-WSL2, amd64), derby, with charset UTF-8.
INFO  2021-11-23 21:13:39,640 [Main Server Thread] com.mirth.connect.server.Mirth: Web server running at http://172.17.0.3:8080/ and https://172.17.0.3:8443/

MySQL 5.7.31 was also running via Docker - and as that worked, I assume that the IP address is fine. If I purposefully change the username or password then I get an error saying that Mirth cant talk to MySQL...therefore when the username/password is correct, and I DONT get an error, I have to assume that I am connected - but the GetTables button doesnt respond with anything.

Mat

**Reply by kpalang** [reply]
There's your problem. Your instance is not connecting to MySQL but the local Derby database.
```
INFO 2021-11-23 21:13:39,639 [Main Server Thread] com.mirth.connect.server.Mirth: Running OpenJDK 64-Bit Server VM 11.0.12 on Linux (5.10.60.1-microsoft-standard-WSL2, amd64), derby, with charset UTF-8.
```
This part `...Linux (5.10.60.1-microsoft-standard-WSL2, amd64), derby...` says you're connected to Derby. You can specify your database with an environment variable `DATABASE`. More info on [Docker Hub](https://hub.docker.com/r/nextgenhealthcare/connect#common-mirth-properties-options) page under `Common mirth.properties options`

**Reply by kpalang** [reply]
Also, for the username and password error. You got that because default Derby connection is credential-less (or just empty values, I'm not 100% sure), so once you provided those authentication wasn't working anymore.

**Comment by matkey** [comment]
I am confused. If I use MySQL 5.7 via Docker then I can read the tables from my MySQL databases using the GetTables button
I dont change Mirth between switching from MySQL 5.7 to 8. I only stop the 5.7 image and start the 8.0 image.

Mirth is setup with all the defaults.

**Reply by matkey** [reply]
I dont see what the database that Mirth uses internally has to do with it - as thats common on both test setups - 5.7 works and 8.0 doesnt

**Reply by pacmano1** [reply]
Firstly, please use code highlighting when you post here (for errors also).
Secondly if you cross post to the forums please reference that in both places with a URL.  Your other post is [https://forums.mirthproject.io/forum/mirth-connect/support/180242-trouble-talking-to-mysql-8#post180271](url)
Thirdly, @kpalang did not understand your question, it happens.    Using Derby however is bad practice.
Fourthly, sometimes "get tables" just does not work in the UI - yet you can still return a result set.  Try it.  Use a real DB tool to explore the schema if needed.   There is a bug with that where you need to put something in the filter.
Fifthly, containers are painful to troubleshoot, make sure you can replicate the problem posted in a non-container set up.
Lastly, I think other folks were fighting JDBC connection strings in mirth for mysql8 , I just set up a clean instance of mysql8 with mirth 3.12 (and configured mysql to allow connections of course) and used ```jdbc:mysql://192.168.66.55:3306/classicmodels?useSSL=tue``` as the URL however I can't show the tables in Mirth unless I put something in the filter.  I can however retrieve data.
